README.md

```diff
@@ -40,13 +42,66 @@ Configure `/etc/readyset_proxysql_scheduler.cnf` as follow:
+* `query_discovery_min_row_sent`: (Optional) - Minimum number of rows sent by a query to be considered a candidate to be cached (Default 0).
+
+# Query Discovery Mode
+The Query Discovery Mode is a set of possible rules to discover queries to automatically cache in Readyset. The options are:
+
+1. `CountStar` - Total Number of Query Executions
+ * Formula: `total_executions = count_star`
+ * Description: This metric gives the total number of times the query has been executed. It is valuable for understanding how frequently the query runs. A high count_star value suggests that the query is executed often.
+
+2. `SumTime` - Total Time Spent Executing the Query
+ * Formula: `total_execution_time = sum_time`
+ * Description: This metric represents the total cumulative time spent executing the query across all its executions. It provides a clear understanding of how much processing time the query is consuming over time. A high total execution time can indicate that the query is either frequently executed or is time-intensive to process.
+
+3. `SumRowsSent` - Total Number of Rows Sent by the Query (sum_rows_sent)
+ * Formula: `total_rows_sent = sum_rows_sent`
+ * Description: This metric provides the total number of rows sent to the client across all executions of the query. It helps you understand the query’s output volume and the amount of data being transmitted. 
+
+4. `MeanTime` - Average Query Execution Time (Mean)
+ * Formula: `mean_time = sum_time / count_star`
+ * Description: The mean time gives you an idea of the typical performance of the query over all executions. It provides a central tendency of how long the query generally takes to execute.
+
+5. `ExecutionTimeDistance` - Time Distance Between Query Executions
+ * Formula: `execution_time_distance = max_time - min_time`
+ * Description: This shows the spread between the fastest and slowest executions of the query. A large range might indicate variability in system load, input sizes, or external factors affecting performance.
+
+6. `QueryThroughput` - Query Throughput
+ * Formula: `query_throughput = count_star / sum_time`
+ * Description: This shows how many queries are processed per unit of time. It’s useful for understanding system capacity and how efficiently the database is handling the queries.
+
+7. `WorstBestCase` - Worst Best-Case Query Performance
+ * Formula: `worst_case = min_time`
+ * Description: The min_time metric gives the fastest time the query was ever executed. It reflects the best-case performance scenario, which could indicate the query’s performance under optimal conditions.
+
+8. `WorstWorstCase` - Worst Worst-Case Query Performance
+ * Formula: `worst_case = max_time`
+ * Description: The max_time shows the slowest time the query was executed. This can indicate potential bottlenecks or edge cases where the query underperforms, which could be due to larger data sets, locks, or high server load.
```

Comment:
Is there an extra use of the word "worst" in here?  I'm not sure what worst best case means.

Reply:
Worst Best Case - We get the queries that have the worst best case, in other words, min_time DESC
Worst Worst Case - we get the queries that have the worst worst case, in other words, max_time DESC

Reply:
Oh, this is talking about more than 1 query?  I think I see now.  I originally read this as a stat for every query, when really it sounds as though this is a state that aggregates over many queries.

Can we amend this description to state somehow that this is talking about multiple queries?

Reply:
I think it should go at:

> The Query Discovery Mode is a set of possible rules to discover queries to automatically cache in Readyset. The options are:

To me this is clear that is meant to indicate the queries filter, but I am biased as I am the author. Do you have any suggestion that will make it more clear ?

Reply:
Hm...  Maybe in the formula?  `worst_case = max(max_time)`?  And for `WorstBestCase`, maybe `worst_case = max(min_time)`?

Alternatively (or additionally), maybe a subtle wording change like this?

Previous:
> The max_time shows the slowest time the query was executed.

Proposed:
> The max_time shows the slowest time **each** query was executed.

I think either of these subtle changes would have helped me read this accurately the first time.

Reply:
> Hm... Maybe in the formula? worst_case = max(max_time)? And for WorstBestCase, maybe worst_case = max(min_time)?

This is what proxysql calculates to us. They measure the execution time of each query and print in max_time the max(execution_time) of that query. But on stats table, there is only the max_time available to us. We just order all the queries based on their max_time descending, which bring the worst first. On think to notice here is that this is internal to the scheduler. There is no information about this being printed to the user. They just know which queries are selected.

Reply:
That makes sense.  When I'm talking about the formula, I'm only saying we can update the documentation to make that more explicit.

It currently says `worst_case = max_time` instead of `worst_case = max(max_time)`.

Reply:
Done